src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -147,52 +148,38 @@ public virtual async Task<int> SendJsonRpcResult(JsonRpcResult result)
                     await enumerator.DisposeAsync();
                 }
 
-                await _handler.SendRawAsync(_jsonClosingBracket, true);
-                singleResponseSize += 1;
-
-                return singleResponseSize;
+                await resultData.WriteAsync(_jsonClosingBracket);
             }
             else
             {
-                return await SendJsonRpcResultEntry(result.SingleResponse!.Value);
+                SendJsonRpcResultEntry(resultData, result.SingleResponse!.Value);
             }
+
+            // ? What if we write more than int.MaxValue.
+            // Result could be negative
+            // return (int)resultData.WrittenBytes;
+            return (int)resultData.WrittenBytes;
         }
 
-        private async Task<int> SendJsonRpcResultEntry(JsonRpcResult.Entry result, bool endOfMessage = true)
+        private void SendJsonRpcResultEntry(Stream dest, JsonRpcResult.Entry result)
         {
-            void SerializeTimeoutException(MemoryStream stream, JsonRpcResult.Entry result)
+            using JsonRpcResult.Entry entry = result;
+
+            try
             {
-                _jsonSerializer.Serialize(stream, _jsonRpcService.GetErrorResponse(
-                    ErrorCodes.Timeout,
-                    "Request was canceled due to enabled timeout.",
-                    result.Response.Id,
-                    result.Response.MethodName));
+                _jsonSerializer.SerializeWaitForEnumeration(dest, result.Response);
```

Comment:
Despite `Stream` supporting async operations, internally we're using Newtonsoft.Json which will only use `System.IO.Stream.Write(byte[] buffer, int offset, int count)`

Reply:
I dont think we're still using Newtonsoft.Json, after Ben's changes? right?

Reply:
nvm I didnt notice PR merged xD